server/src/main/java/org/opensearch/action/admin/cluster/remotestore/repository/RemoteStoreRepositoryRegistrationHelper.java

```diff
@@ -0,0 +1,364 @@
+
+        if (remoteStoreNode) {
+            validateAttributeNonNull(node, REMOTE_STORE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEY);
+            validateAttributeNonNull(node, segmentRepositoryTypeAttributeKey);
+            validateSettingsAttributesNonNull(node, segmentRepositorySettingsAttributeKeyPrefix);
+            validateAttributeNonNull(node, REMOTE_STORE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEY);
+            validateAttributeNonNull(node, translogRepositoryTypeAttributeKey);
+            validateSettingsAttributesNonNull(node, translogRepositorySettingsAttributeKeyPrefix);
+        }
+
+        return remoteStoreNode;
+    }
+
+    private static void compareAttribute(DiscoveryNode joiningNode, DiscoveryNode existingNode, String attributeKey) {
+        String joiningNodeAttributeValue = validateAttributeNonNull(joiningNode, attributeKey);
+        String existingNodeAttribute = validateAttributeNonNull(existingNode, attributeKey);
+
+        if (existingNodeAttribute.equals(joiningNodeAttributeValue) == false) {
+            throw new IllegalStateException(
+                "joining node ["
+                    + joiningNode
+                    + "] has node attribute ["
+                    + attributeKey
+                    + "] value ["
+                    + joiningNodeAttributeValue
+                    + "] which is different than existing node ["
+                    + existingNode
+                    + "] value ["
+                    + existingNodeAttribute
+                    + "]."
+            );
+        }
+    }
+
+    private static void compareSettingsAttributes(DiscoveryNode joiningNode, DiscoveryNode existingNode, String attributeKeyPrefix) {
+        List<String> existingNodeSettingsAttributeKeys = existingNode.getAttributes()
+            .keySet()
+            .stream()
+            .filter(key -> key.startsWith(attributeKeyPrefix))
+            .collect(Collectors.toList());
+        existingNodeSettingsAttributeKeys.stream().forEach(key -> compareAttribute(joiningNode, existingNode, key));
+    }
+
+    /**
+     * During node join request this method will validate if the joining node has the same remote store node
+     * attributes as of the existing node in the cluster.
+     *
+     * TODO: The below check is valid till we support migration, once we start supporting migration a remote
+     *       store node will be able to join a non remote store cluster and vice versa. #7986
+     */
+    public static boolean compareRemoteStoreNodeAttributes(DiscoveryNode joiningNode, DiscoveryNode existingNode) {
+        String segmentRepositoryName = validateAttributeNonNull(existingNode, REMOTE_STORE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEY);
+        String translogRepositoryName = validateAttributeNonNull(existingNode, REMOTE_STORE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEY);
+
+        compareAttribute(joiningNode, existingNode, REMOTE_STORE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEY);
```

Comment:
we are repeating all code for all types of repositories. Can we create a set of repositories instead and do it for all ? This for validating , verifying